### PR TITLE
service: improve tests and service funtions

### PR DESCRIPTION
**Pull request by iancardosozup**
This commits add a ton of tests on service implementation, refactor on some adding more incisive assertions.

Signed-off-by: Ian Cardoso <ian.cardoso@zup.com.br>

**- What I did**

**- How to verify it**

**- Description for the changelog**

### CODE REVIEW

File: internal/services/formatters/service_test.go [hunk 1]
```diff
@@ -364,3 +420,127 @@ func TestService_GetCodeWithMaxCharacters(t *testing.T) {
 		assert.Len(t, newCode, 100)
 	})
 }
+
+//TODO:Does this work on windows?
+func TestRemoveSrcFolderFromPath(t *testing.T) {
+	t.Run("should return path without src prefix", func(t *testing.T) {
+		monitorController := NewFormatterService(&analysis.Analysis{}, &docker.Mock{}, &config.Config{})
+		result := monitorController.RemoveSrcFolderFromPath(filepath.Join("src", "something"))
+		assert.Equal(t, "omething", result)
```
[the remark this answers is not in the document]
Comment (iancardosozup) [reply internal/services/formatters/service_test.go:429]: actually not, i tought it was but it's a bug, the function should return `something` but returns `omething` instead, i will fix it.

File: internal/services/formatters/service.go [hunk 2]
```diff
@@ -187,21 +183,10 @@ func (s *Service) truncatedCode(code string, column int) string {
...
-
-func (s *Service) GetProjectPathWithWorkdir(projectSubPath string) string {
-	if projectSubPath != "" && projectSubPath[0:1] == string(os.PathSeparator) {
-		return fmt.Sprintf("%s%s", s.GetConfigProjectPath(), projectSubPath)
-	}
+	basePath := filepath.Join(s.GetConfigProjectPath(), projectSubPath)
+	filepathWithFileName := file.GetPathFromFilename(filename, basePath)
 
-	return fmt.Sprintf("%s%s%s", s.GetConfigProjectPath(), string(os.PathSeparator), projectSubPath)
+	return path.Join(projectSubPath, filepathWithFileName)
```
Comment (matheusalcantarazup) [inline internal/services/formatters/service.go:189]: Change to `filepath.Join` instead. The [path](https://pkg.go.dev/path) package does not deal with Windows paths.

File: internal/services/formatters/service.go [hunk 3]
```diff
@@ -253,17 +236,9 @@ func (s *Service) newVulnerabilityFromFinding(finding *engine.Finding, tool tool
 	}
 }
 
-func (s *Service) removeHorusecFolder(filepath string) string {
+func (s *Service) removeHorusecFolder(pathFromFile string) string {
```
Comment (matheusalcantarazup) [inline internal/services/formatters/service.go:239]: Same ideia from comment above about the arg name.

File: internal/services/formatters/service_test.go [hunk 4]
```diff
@@ -364,3 +421,129 @@ func TestService_GetCodeWithMaxCharacters(t *testing.T) {
...
+		assert.Equal(t, filepath.Join(relativeFilePath, filename), result)
+		t.Cleanup(func() {
+			_ = file.Close()
+			_ = os.RemoveAll(dirName)
+		})
+
+	})
+	t.Run("should return empty when path from filename is not found", func(t *testing.T) {
+		cfg := &config.Config{}
+		cfg.ProjectPath = testutil.GoExample1
```
Comment (matheusalcantarazup) [inline internal/services/formatters/service_test.go:473]: Fill the fields directly as above.

File: internal/services/formatters/service_test.go [hunk 5]
```diff
@@ -364,3 +421,129 @@ func TestService_GetCodeWithMaxCharacters(t *testing.T) {
...
+
+		result := monitorController.GetFilepathFromFilename(filename, "")
+
+		assert.Equal(t, "", result)
+	})
+}
+
+func TestGetConfigCMDByFileExtension(t *testing.T) {
+	t.Run("should return path when valid parameters", func(t *testing.T) {
+		cfg := &config.Config{}
```
Comment (matheusalcantarazup) [inline internal/services/formatters/service_test.go:485]: Same here.

**Review by matheusalcantarazup — CHANGES_REQUESTED** [review]

**Review by matheusalcantarazup — APPROVED** [review]

**Pull request merged**